contracts/feature-tests/basic-features/src/managed_buffer_features.rs

```diff
@@ -57,7 +57,21 @@ pub trait ManagedBufferFeatures {
     }
 
     #[endpoint]
-    fn mbuffer_slice_2(
+    fn mbuffer_set_slice(
+        &self,
+        mb: ManagedBuffer,
+        index: usize,
+        item: &[u8],
+    ) -> SCResult<ManagedBuffer> {
+        let mut result = mb;
+        result
+            .set_slice(index, item)
+            .map_err(|_| "index out of bounds")?;
+        Ok(result)
```

Comment:
Wouldn't the following have been enough?
```suggestion
        result.set_slice(index, item)?
```
Or to the error types differ?